.github/workflows/docs.yml

```diff
@@ -0,0 +1,40 @@
+
+name: Documentation
+on:
+  push:
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    container:
+      image: 'archlinux:latest'
```

Comment:
I'm confused: why do we have `ubuntu-20.04` above and `archlinux:latest` here?

Reply:
ubuntu-20.04 is the host on which `archlinux:latest`, a docker container runs. There are options on which host can be chosen:
* ubuntu-16.04
* ubuntu-18.04
* ubuntu-20.04
* windows-latest
* macos-latest
Ubuntu is the only linux based VM available. And then we use docker to pull `archlinux:latest`

Reply:
To simplify a bit more, those lines expand to:
```bash

# set up the VM
az create path/to/vm/config   # set up a VM on Azure (because, its Microsoft stuff, implicit, done by GitHub interally)

ssh runner@$VM_IP_ADDR  # internal VM IP Address >>

sudo apt-get install docker  # iirc

docker run -it archlinux:latest 
```